File: 0013-roman-to-integer/0013-roman-to-integer.cs
public class Solution {
    public int RomanToInt(string s) {
        Dictionary<char, int> myDictionary = new Dictionary<char, int>();
        myDictionary['I'] = 1;
        myDictionary['V'] = 5;
        myDictionary['X'] = 10;
        myDictionary['L'] = 50;
        myDictionary['C'] = 100;
        myDictionary['D'] = 500;
        myDictionary['M'] = 1000;

        int result = myDictionary[s[^1]];
        for (int i = s.Length - 2; i >= 0; --i) {
            if (myDictionary[s[i]] < myDictionary[s[i + 1]]) {
                result -= myDictionary[s[i]];
            } else {
                result += myDictionary[s[i]];
            }
        }
        return result;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        my_map = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}

        result = my_map[s[-1]]
        for i in range(len(s) - 2, -1, -1):
            if my_map[s[i]] < my_map[s[i + 1]]:
                result -= my_map[s[i]]
            else:
                result += my_map[s[i]]
        
        return result